firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/persistence/CrashlyticsReportPersistence.java

```diff
@@ -43,36 +43,41 @@
 
   private static final Charset UTF_8 = Charset.forName("UTF-8");
 
-  private static final String WORKING_DIRECTORY_NAME = "fl";
+  private static final String WORKING_DIRECTORY_NAME = "report-persistence";
   private static final String OPEN_SESSIONS_DIRECTORY_NAME = "sessions";
-  private static final String FATAL_DIRECTORY_NAME = "fatal";
-  private static final String NON_FATAL_DIRECTORY_NAME = "non-fatal";
+  private static final String PRIORITY_REPORTS_DIRECTORY = "priority-reports";
+  private static final String REPORTS_DIRECTORY = "reports";
 
-  private static final String REPORT_FILE_NAME = "report.json";
+  private static final String REPORT_FILE_NAME = "report";
   private static final String EVENT_FILE_NAME_PREFIX = "event";
-  private static final String EVENT_FILE_NAME_FORMAT = EVENT_FILE_NAME_PREFIX + "%s.json";
-  private static final String EVENT_COUNTER_FORMAT = "%010d";
-
-  private static final String EVENT_TYPE_FATAL = "crash";
+  private static final int EVENT_COUNTER_WIDTH = 10; // String width of maximum positive int value
+  private static final String EVENT_COUNTER_FORMAT = "%0" + EVENT_COUNTER_WIDTH + "d";
+  private static final int EVENT_NAME_LENGTH =
+      EVENT_FILE_NAME_PREFIX.length() + EVENT_COUNTER_WIDTH;
+  private static final String PRIORITY_EVENT_SUFFIX = "_";
+  private static final String NORMAL_EVENT_SUFFIX = "";
 
   private static final CrashlyticsReportJsonTransform TRANSFORM =
       new CrashlyticsReportJsonTransform();
 
   private final AtomicInteger eventCounter = new AtomicInteger(0);
```

Comment:
What resets this back to zero? Is this "Persistence" class recreated before it gets sufficiently large?

Reply:
In practice, yes. If you've got a situation where you have greater than `Integer.MAX_VALUE` events coming in for a single session, something is terribly wrong, and the information isn't going to be particularly useful to you anyway.

See also: https://github.com/firebase/firebase-android-sdk/blob/master/firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/core/CrashlyticsController.java#L241